src/connection/manager.ts

```diff
@@ -86,11 +87,11 @@ export class SQLJobManager {
     return (this.selectedJob >= 0);
   }
 
-  runSQL<T>(query: string): Promise<T[]> {
+  async runSQL<T>(query: string): Promise<T[]> {
     const selected = this.jobs[this.selectedJob]
     if (SQLJobManager.jobSupport && selected) {
-      return selected.job.query(query) as Promise<T[]>;
-
+      let results = await selected.job.query<T>(false, query).run();
```

Comment:
We need to call `run` with a high value here.

Reply:
Resolved at https://github.com/halcyon-tech/vscode-db2i/pull/23/commits/c8fc13db33e849e64ca40f67982004830148b867